### PR TITLE
Add init command

### PR DESCRIPTION
**Pull request by frelon**
The init command is meant to be run in a container build step and extracts config-files needed for an elemental derivative.

This PR includes a way to turn on/off different features of the elemental-toolkit. The standard features included here are:
* immutable-rootfs
* grub-config
* elemental-setup
* dracut-config
* cloud-config-defaults
* cloud-config-essentials

An example dockerfile:

```
FROM rancher/elemental-cli:latest AS elemental-cli
FROM opensuse/tumbleweed as default

RUN zypper install -y grub2 grub2-x86_64-efi grub2-x86_64-efi-extras shim dracut kernel kernel-firmware-all systemd \
                      bash syslinux lvm2 parted dosfstools e2fsprogs rsync util-linux-systemd coreutils \
                      squashfs NetworkManager device-mapper iproute2 tar curl ca-certificates ca-certificates-mozilla \
                      procps openssl openssh vim-small less iputils bind-utils wget jq

COPY --from=elemental-cli /usr/bin/elemental /usr/bin/elemental

RUN systemctl enable NetworkManager
RUN elemental init --force immutable-rootfs,grub-config,dracut-config
```

Fixes rancher/elemental-toolkit#1766

### CODE REVIEW

**Review by fgiudici — APPROVED** [review]
Great PR! 
created a derivative from tumbleweed with the elemental init cmd 🎉

**Review by davidcassany — CHANGES_REQUESTED** [review]
Looks good, I just have few doubts about some details.

Should be easy to arrange, probably it is already fine and I just missed something.

I think we also should adapt the specfile of the elemental-toolkit RPM so we can merge it being transparent to elemental teal. This way we could merge without necessarily breaking `Dev` and apply the transition in Elemental Teal in smaller PRs.

File: pkg/action/init.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		if err := feature.Install(cfg.Config.Logger, cfg.Config.Fs, cfg.Config.Runner); err != nil {
+			cfg.Config.Logger.Errorf("Error installing feature '%s': %v", feature.Name, err.Error())
+			return err
+		}
+	}
+
+	cfg.Config.Logger.Info("Setting GRUB nextboot")
+
+	grub := utils.NewGrub(&cfg.Config)
+	firstboot := map[string]string{"next_entry": "recovery"}
```
[the remark this answers is not in the document]
Comment (frelon) [reply pkg/action/init.go:58]: Yep, this was copied from the old `grub2-config` luet package, maybe we should just drop it?

[the remark this answers is not in the document]
Comment (frelon) [reply pkg/action/init.go:58]: I removed it :+1:

File: pkg/features/features.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+var files embed.FS
+
+const (
+	embeddedRoot = "embedded"
+
+	FeatureImmutableRootfs = "immutable-rootfs"
+	FeatureGrubConfig      = "grub-config"
+	FeatureElementalSetup  = "elemental-setup"
+	FeatureDracutConfig    = "dracut-config"
+	FeatureCloudConfig     = "cloud-config"
```
[the remark this answers is not in the document]
Comment (frelon) [reply pkg/features/features.go]: Good idea! I split cloud-config into cloud-config-defaults and cloud-config-essentials, where the essentials are the services used in teal.

**Comment by frelon** [comment]
@davidcassany I did the changes to obs to keep Dev in a working condition and split the `cloud-config` feature into two, let me know what you think! :+1:

**Pull request merged**